### ISSUE TITLE
cannot generate an address containing no shielded receivers.

### ISSUE THREAD

**Issue by gades** [zcash/zcash#6421]
### Describe the issue
After disabling getnewaddress I can't use p2pkh anymore

#### If so, please list the steps to reproduce below:
1. zcash-cli z_getaddressforaccount 1 '["p2pkh"]'

### Expected behaviour
I'm expecting new transpired address p2pkh

### Actual behaviour + errors
$ zcash-cli z_getaddressforaccount 1 '["p2pkh"]'
error code: -4
error message:
Error: cannot generate an address containing no shielded receivers.

### The version of Zcash you were using:
Zcash Daemon version v5.4.0

**Comment by str4d** [comment]
This is expected behaviour. Unified addresses are required to contain a shielded receiver. You can extract the P2PKH receiver from a unified address with `z_listunifiedreceivers`.

Can you describe how you currently use `zcashd`? We generally want to migrate users away from the legacy Bitcoin Core-inherited transparent-only APIs, and we want to ensure that transparent-only use cases can be migrated to use unified addresses with shielded components.

**Comment by gades** [comment]
It's exchange and I had best way with getnewaddress.  Will you support old addresses in future ? if yes, why you decide to remove getnewaddress because it steel available in bitcoin core and others.
Also, when zchash core will support wallet encryption?

WARNING: encryptwallet is disabled.
To enable it, restart zcashd with the following command line options:
-experimentalfeatures and -developerencryptwallet

**Comment by gituser** [comment]
I second this question, we rely on `getnewaddress` to generate new users deposits. If you remove `getnewaddress` we will no longer support zcash. 

What is the reason for the removal?

**Comment by gituser** [comment]
@got3nks what is the reason for removal of `z_getnewaddress` ?
Why not make it simple to generate  new address for zcash.

**Comment by gades** [comment]
> > I second this question, we rely on `getnewaddress` to generate new users deposits. If you remove `getnewaddress` we will no longer support zcash.
> > What is the reason for the removal?
> 
> From my understanding **transaparent** addresses will still be available even after Stage 3 deprecation of `getnewaddress`?
> 
> 1. Generate an unified address through `z_getaddressforaccount`
> 2. Call `z_listunifiedreceivers` with the generated unified address to get the **p2pkh** _transparent_ address

I don't know why one simple RPC getnewaddress is going to be deprecated and the we must use two RCP calls, why?

**Comment by gituser** [comment]
I agree with @gades why remove something that's been working for quite long for users and replace with new untested RPCs, possibly resulting in more bugs (e.g. parallel address requests or some other scenarios).

**Comment by gades** [comment]
I did request #6444 for make an alternative rpc function , please vote for it and probably they will decide to don't remove old one
